### PR TITLE
feat(VirtualizedList): adapt and enhance component objects

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
With virtualizedList, the component object List is no longer up to date.

**What is the chosen solution to this problem?**
Adapt it, but not only : 
* enhance its use to have a more fluent api
* add missing selectors

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**
The List api and POJO have completely changed.
See breaking change log

### CODE REVIEW

File: packages/components/e2e/component-objects/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.talend</groupId>
     <artifactId>component-objects</artifactId>
-    <version>1.0.0</version>
+    <version>0.95.0-SNAPSHOT</version>
```
Comment (frassinier) [inline packages/components/e2e/component-objects/pom.xml]: Are you sure that it will be easily to bump?

Comment (jsomsanith) [inline packages/components/e2e/component-objects/pom.xml]: I discussed with @kmouzouri and @atriki from TFD team, they are the only ones using the lib, and they don't maintain any (release) branch for now. So it's the time to align version and override the 1.0.0.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java [hunk 2]
```diff
@@ -58,6 +75,11 @@ public Component(WebDriver driver, String name, String selector) {
      */
     public WebElement getElement() throws NotFoundException {
         LOGGER.info(this.name + ".getElement " + this.selector);
+
+        if(this.root != null) {
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: Missing space before if.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java [hunk 3]
```diff
@@ -41,6 +43,21 @@ public Component(WebDriver driver, String name, String selector) {
...
+     * @param driver Selenium WebDriver
+     * @param root Component root element
+     */
+    public Component(WebDriver driver, String name, WebElement root) {
+        LOGGER.info("Component " + name + " from provided root");
+        this.driver = driver;
+        this.name = name;
+        this.root = root;
+
+        this.driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: Why the timeout?

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        this(driver, null);
+    }
+
+    /**
+     * Constructor with table id
+     *
+     * @param driver Selenium WebDriver
+     */
+    public Table(final WebDriver driver, final String id) {
+        super(driver, NAME, String.format(TABLE_SELECTOR, id != null ? "#" + id : ""));
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java]: Please set a default value for id instead of using a ternary.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    /**
+     * Get all rendered items, represented by a row in the table.
+     *
+     * @return The list of Items
+     */
+    public List<Item> getItems() {
+        return this.getElement() //
+                .findElements(By.cssSelector(TABLE_ITEM_SELECTOR)) //
+                .stream() //
+                .map(webElement -> new Item(driver, webElement)) //
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java]: Those comments slashes are not required.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Item.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+/**
+ * A List is used to easy access to WebElements of the react-talend-component List component - Table - Item.
+ */
+public class Item extends Component {
+
+    private static final String NAME = "Item";
+
+    private static final String TABLE_ITEM_TITLE_SELECTOR = ".tc-list-title > button";
+
+    private static final String TABLE_ITEM_ACTIONS_SELECTOR = ".tc-list-title > .tc-actions button";
```
Comment (frassinier) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Item.java]: DRY, create var for `.tc-list-title`

File: packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
+package org.talend.component;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class StorybookTest {
```
Comment (frassinier) [inline packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java]: What's about run this test into Travis CI?

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java (hunk 3 above)
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: This was set by the TFD QA, I just kept it.
This is their settings. But now that you point it, it should not be there, it overrides the app test config.
I'll remove it.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java (hunk 4 above)
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java]: I can't, no default value in methode prototype in java.
Plus, super() must be the first instruction. So no choice :(

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java (hunk 5 above)
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java]: If I don't put them, and format it with Talend config, it is less readable.
That's why I choose to put them.

File: packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java (hunk 7 above)
Comment (jsomsanith) [inline packages/components/e2e/component-objects/src/test/java/org/talend/component/StorybookTest.java]: :+1: let's do that in another PR

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java (hunk 5 above)
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java]: :roll_eyes:
